src/pages/SignUp.tsx

```diff
@@ -0,0 +1,537 @@
+        backgroundRepeat: 'no-repeat',
+        ...theme.applyStyles('dark', {
+            backgroundImage:
+                'radial-gradient(at 50% 50%, hsla(210, 100%, 16%, 0.5), hsl(220, 30%, 5%))',
+        }),
+    },
+}));
+
+{
+    /* Checking functions to make sure all inputs are valid */
+}
+export default function SignUp(props: { disableCustomTheme?: boolean }) {
+    const [firstName, setFirstName] = useState('');
+    const [firstNameError, setFirstNameError] = useState(false);
+    const handleChangeFirstName = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setFirstName(e.target.value);
+    };
+    const handleBlurFirstName = () => {
+        setFirstNameError(!firstName.length ? true : false);
+    };
+
+    const [lastName, setLastName] = useState('');
+    const [lastNameError, setLastNameError] = useState(false);
+    const handleChangeLastName = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setLastName(e.target.value);
+    };
+    const handleBlurLastName = () => {
+        setLastNameError(!lastName.length ? true : false);
+    };
+
+    const [email, setEmail] = useState('');
+    const [emailError, setEmailError] = useState(false);
+    const handleChangeEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setEmail(e.target.value);
+    };
+    const handleBlurEmail = () => {
+        if (!validateEmail(email)) {
+            setEmailError(true);
+        } else {
+            setEmailError(false);
+        }
+    };
+
+    const [password, setPassword] = useState('');
+    const [showPassword, setShowPassword] = useState(false);
+    const [passwordError, setPasswordError] = useState(false);
+    const handleChangePassword = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setPassword(e.target.value);
+    };
+    const handleBlurPassword = () => {
+        if (!validatePassword(password)) {
+            setPasswordError(true);
+        } else {
+            setPasswordError(false);
+        }
+    };
+
+    const [reenterPassword, setReenterPassword] = useState('');
+    const [reenterPasswordError, setReenterPasswordError] = useState(false);
+    const handleChangeReenterPassword = (
+        e: React.ChangeEvent<HTMLInputElement>
+    ) => {
+        setReenterPassword(e.target.value);
+        handleBlurReenterPassword();
+    };
+    const handleBlurReenterPassword = () => {
+        if (password !== reenterPassword) {
+            setReenterPasswordError(true);
+        } else {
+            setReenterPasswordError(false);
+        }
+    };
+
+    const togglePasswordVisibility = () => {
+        setShowPassword((prev) => !prev);
+    };
+
+    const validateEmail = (email: string) => {
+        return emailRegex.test(email);
+    };
+
+    const validatePassword = (password: string) => {
+        return passwordRegex.test(
+            password
+        );
+    };
+
+    const validateForm = () => {
```

Comment:
I was able to register an account without having the passwords match. This function should probably check that the password fields match.